File: AndroidApp/src/main/java/com/smp/sharedkotlin/app/data/MemoRepositoryImpl.kt
package com.smp.sharedkotlin.app.data

import com.smp.sharedkotlin.domain.MemoRepository
import com.smp.sharedkotlin.entity.MemoEntity


class MemoRepositoryImpl(private val dataSource: MemoDataSource) : MemoRepository {

    override fun getAllMemos(): List<MemoEntity> {
        return MemoEntityMapper.mapAll(dataSource.getAllMemos())
    }

    override fun getMemo(id: Int): MemoEntity? {
        dataSource.getMemo(id)?.let {
            return MemoEntityMapper.map(it)
        }
        return null
    }
}

File: iOSApp/iOSApp/Data/MemoRepositoryImpl.swift
import SharedCode

class MemoRepositoryImpl: NSObject, MemoRepository {
    
    let mapper = MemoEntityMapper()
    
    let dataSource: MemoDataSource
    
    init(dataSource: MemoDataSource) {
        self.dataSource = dataSource
    }
    
    func getAllMemos() -> [MemoEntity] {
        return mapper.mapAll(objs: dataSource.getAllMemos())
    }
    
    func getMemo(id: Int32) -> MemoEntity? {
        guard let memo = dataSource.getMemo(id: id) else {
            return nil
        }
        return mapper.map(obj: memo)
    }
}
